scapy/contrib/automotive/gm/gmlanutils.py

```diff
@@ -0,0 +1,384 @@
+from scapy.utils import PeriodicSenderThread
+
+
+__all__ = ["GMLAN_TesterPresentSender", "GMLAN_InitDiagnostics",
+           "GMLAN_GetSecurityAccess", "GMLAN_RequestDownload",
+           "GMLAN_TransferData", "GMLAN_TransferPayload",
+           "GMLAN_ReadMemoryByAddress", "GMLAN_BroadcastSocket"]
+
+log_loading.info("\"conf.contribs['GMLAN']"
+                 "['treat-response-pending-as-answer']\" set to True). This "
+                 "is required by the GMLAN-Utils module to operate "
+                 "correctly.")
+try:
+    conf.contribs['GMLAN']['treat-response-pending-as-answer'] = False
+except KeyError:
+    conf.contribs['GMLAN'] = {'treat-response-pending-as-answer': False}
+
+
+class GMLAN_TesterPresentSender(PeriodicSenderThread):
+    def __init__(self, sock, pkt=GMLAN(service="TesterPresent"), interval=2):
+        """ Thread to send TesterPresent messages packets periodically
+
+        Args:
+            sock: socket where packet is sent periodically
+            pkt: packet to send
+            interval: interval between two packets
+        """
+        PeriodicSenderThread.__init__(self, sock, pkt, interval)
+
+
+def GMLAN_InitDiagnostics(sock, broadcastsocket=None, timeout=None,
+                          verbose=None, retry=0):
+    """Send messages to put an ECU into an diagnostic/programming state.
+
+    Args:
+        sock:       socket to send the message on.
+        broadcast:  socket for broadcasting. If provided some message will be
+                    sent as broadcast. Recommended when used on a network with
+                    several ECUs.
+        timeout:    timeout for sending, receiving or sniffing packages.
+        verbose:    set verbosity level
+        retry:      number of retries in case of failure.
+
+    Returns true on success.
+    """
+    if verbose is None:
+        verbose = conf.verb
+    retry = abs(retry)
+
+    while retry >= 0:
+        retry -= 1
+
+        # DisableNormalCommunication
+        p = GMLAN(service="DisableNormalCommunication")
+        if broadcastsocket is None:
+            if verbose:
+                print("Sending %s" % repr(p))
+            resp = sock.sr1(p, timeout=timeout, verbose=0)
+            if resp is not None:
+                if verbose:
+                    resp.show()
+                if resp.sprintf("%GMLAN.service%") != "DisableNormalCommunicationPositiveResponse":   # noqa: E501
+                    continue
+            else:
+                if verbose:
+                    print("Timeout.")
+                continue
+        else:
+            if verbose:
+                print("Sending %s as broadcast" % repr(p))
+            broadcastsocket.send(p)
+        time.sleep(0.05)
+
+        # ReportProgrammedState
+        p = GMLAN(service="ReportProgrammingState")
+        if verbose:
+            print("Sending %s" % repr(p))
+        resp = sock.sr1(p, timeout=timeout, verbose=0)
+        if resp is not None:
```

Comment:
you could remove on level of indentation:
```python
        if resp is None:
            if verbose:
                print("Timeout.")
            continue
        if verbose:
            resp.show()
        if resp.sprintf("%GMLAN.service%") == "NegativeResponse":
            continue
```

I think this would be possible on all your commands